doc/languages-frameworks/go.section.md

```diff
@@ -142,6 +142,7 @@ Many attributes [controlling the build phase](#variables-controlling-the-build-p
 - [`patchFlags`](#var-stdenv-patchFlags)
 - [`postPatch`](#var-stdenv-postPatch)
 - [`preBuild`](#var-stdenv-preBuild)
+- `env`
```

Comment:
I think we should add at least a sentence here to explain things

Reply:
@SuperSandro2000 i think given context, this is adequate,  I had the same first reaction myself

Reply:
You do have a point, the other attrs are mostly self-explanatory while `env` might be a bit confusing as to why it's needed.
I added a short sentence to mention `GOWORK`.